File: frontend/temporal-trace-ui/src/app/models/task-branch.ts
export interface TaskBranch {
  branchId: string;
  taskId: number;
  branchName: string;
  createdFromTime: string;
  createdAt: string;
  isMainTimeline: boolean;
}

export interface BranchTimeline {
  branchId: string;
  branchName: string;
  isMainTimeline: boolean;
  taskSnapshot?: any;
}

export interface CreateBranchRequest {
  targetTime: string;
  branchName: string;
}

File: backend/TemporalTrace.Api/Models/TaskBranch.cs
namespace TemporalTrace.Api.Models;

public class TaskBranch
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public int TaskId { get; set; }
    public string BranchName { get; set; } = string.Empty;
    public DateTime CreatedFromTime { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public bool IsMainTimeline { get; set; } = false;
}
